command/agent/dns.go

```diff
@@ -547,17 +566,17 @@ RPC:
 	}
 
 	// If the network is not TCP, restrict the number of responses
-	if network != "tcp" && len(resp.Answer) > maxServiceResponses {
-		resp.Answer = resp.Answer[:maxServiceResponses]
+	if network != "tcp" {
+		wasTrimmed := trimAnswers(resp)
```

Comment:
It's slightly confusing, but `EnableTruncate` only determines whether the truncate bit is set, not whether truncation happens.

> [When the DNS query is performed using UDP, Consul will truncate the results without setting the truncate bit. This is to prevent a redundant lookup over TCP that generates additional load. If the lookup is done over TCP, the results are not truncated.](https://www.consul.io/docs/agent/dns.html)